src/parser/parser.ts

```diff
@@ -181,6 +181,7 @@ export class TemplateRoot {
     public readonly dirname: string;
     public readonly filename: string;
     public readonly filepath: string;
+    public readonly developmentRole: boolean;
```

Comment:
I would keep this boolean value of the the TemplateRoot. I would store it in a static variable instead.
whether running in --dev mode or not is something that doesn't neccesarily apply to a template, it applies to the process as a whole. this also means that you dont have to add it to the TemplateRoot constructor and pass it around in the other places